### DISCUSSION TITLE
Cachet 2.x Maintenance

### DISCUSSION THREAD

**Discussion by jbrooksuk** [cachethq/cachet#4350] [Polls]
When I took back Cachet I made the decision to stop supporting 2.x while I worked on 3.x. This was for a couple of reasons, but mainly that I don’t personally have the time to work on 3.x and support an outdated version alongside that.

However, I’ve been listening to your feedback and I do think that we should at least provide some bug / security fixes to 2.x. There should be no new features on 2.x as it’d soon become a game of cat and mouse.

If we agree that we should be supporting 2.x and I can find one or two volunteers who will work on the 2.x branch to patch and release bug fixes, then I can get behind that.

Are you able to help? Please let me know!

**Comment by DennisSkov** [comment]
I've voted "yes" but it's only a small "yes".
Once v3 is released, our plan is to upgrade. If v3 is not ready for a release in the foreseeable future, we would appreciate bug fixes for v2!

**Reply by jbrooksuk** [reply]
v3 is not going to be ready for at least a couple of months! Though, the more people that are able to [help out](https://github.com/orgs/cachethq/discussions/4349), the better.

**Comment by roblevesque** [comment]
I also voted yes, but agree with DennisSkov above me. I am trying to rebuild our deployment on a new VM with a supported OS (Currently on Ubuntu 18.04 and my infosec director is getting noisy about it) and the two deployment methods (direct install and Docker) are failing for me. Some bug fixes and security improvements in the meantime while waiting for 3.x would be appreciated. 

However, once 3.x is out, I plan on re-deploying a new VM anyway. 

I wish I could offer some development assistance, but I don't have experience in PHP. I might be able to assist elsewhere though.

**Reply by jbrooksuk** [reply]
Understandable! Ubuntu 18.04 is now EOL. You _can_ run Cachet on Ubuntu 22.04 (which we use for the demo.cachethq.io instance).

**Reply by warwickw** [reply]
PHP 8 or did you back install PHP7 ?

**Comment by LrkDev** [comment]
I just discovered Cachet because I was looking into monitoring solutions for my small usage, and right away learned about the new 3.x version.
I'm pretty busy right now, but I'll try to follow the project evolution.

If you decide to support the old version and I can make time for it, it would be a pleasure to help if there are some issues I'm able to solve.

**Reply by jbrooksuk** [reply]
Welcome to Cachet! I'd love the support, if we go down this route. Thanks!

**Comment by TheDevMinerTV** [comment]
Please bring back the old documentation for v2, it's gone from GitHub and the new VitePress-based documentation is missing a lot of stuff (and sometimes doesn't render properly).

**Reply by jbrooksuk** [reply]
What rendering issues are you getting?

**Reply by jbrooksuk** [reply]
Can you open an issue on the [cachethq/docs](/cachethq/docs) repo, please.

**Comment by rive-n** [comment]
This is not an easy question at all. Especially if you are the only developer.  
During my initial research I found that the codebase of the v2 is quite clean, but there are a lot of old packages. 

So, as for me I think that you can maintain only some `LATEST` versions, fix some bugs and security issues while working on new version. I know that a lot of other companies using this sort of strategy. With this strategy you will `update` some v2 versions, fixing some reports, but not adding some new functionality to it because people need to understand that new functionality will appear in v3 version for sure.

**Comment by gramakri** [comment]
I voted yes but to elaborate. I wanted to deploy Cachet but v2 was not maintained and v3 was not ready. I think just having v2 maintained (just updating libraries and making periodic releases is good enough) will help show that the project is alive. v3 then can take even a year, it's fine. Of course, when v3 is out, I will upgrade.

**Comment by warwickw** [comment]
Does v2.4.1  support PHP 8 or still 7 only ?

**Reply by jbrooksuk** [reply]
Yes, you can run v2.4 on PHP 7.3 - this is what the demo is using.
